tools/importer/cmsis_imorter.py

```diff
@@ -0,0 +1,143 @@
+    files = os.listdir(folder)    
+    for file in files:
+        abs_src_file = os.path.join(folder, file)
+        if os.path.isfile(abs_src_file):
+            abs_dst_file = os.path.join(path, file)
+            copy_file(abs_src_file, abs_dst_file)
+    
+def run_cmd(command, exit_on_failure=False):
+    """ Passes a command to the system and returns a True/False result once the 
+        command has been executed, indicating success/failure. Commands are passed
+        as a list of tokens. 
+        E.g. The command 'git remote -v' would be passed in as ['git', 'remote', '-v']
+    """
+    return_code = subprocess.call(command, shell=True)
+    
+    if return_code:
+        print("The command %s failed with return code: %s"
+                        %(' '.join(command), return_code))
+        if exit_on_failure:
+            sys.exit(1)
+    
+    return return_code
+
+def remove_repo(folder):
+    os.chdir(abspath(dirname(__file__)))
+    if os.path.exists(folder):
+        rmtree(folder)
+
+def get_repo(repo, branch, folder):
+    """ Get the Repository files from git, at depth level 1
+        repo - Git repository link
+        branch - repository branch
+        folder - folder at which repo will be cloned
+    """
+    remove_repo(folder)
+    clone_cmd = ['git', 'clone', repo, "-b", branch, "--depth", '1', folder]
+    run_cmd(clone_cmd, exit_on_failure=True)
+  
+if __name__ == "__main__":
+    
+    # Read configuration data
+    with open(os.path.join(os.path.dirname(__file__), "cmsis_importer.json"), 'r') as config:        
+        json_data = json.load(config)
+    config = json_data["config"]
+    cmsis_repo = config['cmsis_repo']
+    cmsis_branch = config['cmsis_branch']    
+    data_files = json_data["files"]
+    data_folders = json_data["folders"]
+    
+    print "Fetching git repo"
+    get_repo(cmsis_repo, cmsis_branch, CMSIS_REPO)
+    
+    ## Remove all files listed in .json from mbed-os repo to avoid duplications
+    print "Cleaning up:"
+    for file in data_files:
+        cmsis_file =  file['cmsis_file']
+        del_file(os.path.basename(cmsis_file))
+    
+    for folder in data_folders:
+        cmsis_folder =  os.path.join(CMSIS_PATH, folder['cmsis_folder'])
+        files = os.listdir(cmsis_folder)        
+        for file in files:            
+            del_file(os.path.basename(file))
+
+    ## Copy all the CMSIS files listed in json file to mbed-os    
+    print "Files Copied:"    
+    for file in data_files:        
+        cmsis_file =  os.path.join(CMSIS_PATH, file['cmsis_file'])
+        mbed_path =  os.path.join(ROOT, file['mbed_file'])
+        copy_file(cmsis_file, mbed_path)
+
+    for folder in data_folders:
+        cmsis_folder =  os.path.join(CMSIS_PATH, folder['cmsis_folder'])
+        mbed_path =  os.path.join(ROOT, folder['mbed_folder'])        
+        copy_folder(cmsis_folder, mbed_path)
+
+    #Remove CMSIS Repo
+    remove_repo(CMSIS_REPO)
```

Comment:
We don't want to clone it so we don't want to delete it.

Reply:
We need to commit CMSIS file changes before cherry-pick is performed. If we do not clean CMSIS repo then it will be part of commit. I am fine with not cloning and picking up as command line.
But if we clone we will have to remove it.
Also, as part of command line option it should be outside mbed-os repo.

Reply:
Let's not clone it and add a line in docs mentioning it has to be outside.